src/test/pow_tests.cpp

```diff
@@ -11,112 +11,40 @@
-    blockIndexPrev8.pprev = &blockIndexPrev9;
-    CBlockIndex blockIndexPrev10 = CBlockIndex();
-    blockIndexPrev10.nTime = 1408729587;  // Block #123446
-    blockIndexPrev10.nBits = 0x1b10e878;
-    blockIndexPrev9.pprev = &blockIndexPrev10;
-    CBlockIndex blockIndexPrev11 = CBlockIndex();
-    blockIndexPrev11.nTime = 1408729576;  // Block #123445
-    blockIndexPrev11.nBits = 0x1b1063d0;
-    blockIndexPrev10.pprev = &blockIndexPrev11;
-    CBlockIndex blockIndexPrev12 = CBlockIndex();
-    blockIndexPrev12.nTime = 1408729474;  // Block #123444
-    blockIndexPrev12.nBits = 0x1b104297;
-    blockIndexPrev11.pprev = &blockIndexPrev12;
-    CBlockIndex blockIndexPrev13 = CBlockIndex();
-    blockIndexPrev13.nTime = 1408729305;  // Block #123443
-    blockIndexPrev13.nBits = 0x1b107556;
-    blockIndexPrev12.pprev = &blockIndexPrev13;
-    CBlockIndex blockIndexPrev14 = CBlockIndex();
-    blockIndexPrev14.nTime = 1408729179;  // Block #123442
-    blockIndexPrev14.nBits = 0x1b110764;
-    blockIndexPrev13.pprev = &blockIndexPrev14;
-    CBlockIndex blockIndexPrev15 = CBlockIndex();
-    blockIndexPrev15.nTime = 1408729116;  // Block #123441
-    blockIndexPrev15.nBits = 0x1b1141bf;
-    blockIndexPrev14.pprev = &blockIndexPrev15;
-    CBlockIndex blockIndexPrev16 = CBlockIndex();
-    blockIndexPrev16.nTime = 1408728950;  // Block #123440
-    blockIndexPrev16.nBits = 0x1b1123f9;
-    blockIndexPrev15.pprev = &blockIndexPrev16;
-    CBlockIndex blockIndexPrev17 = CBlockIndex();
-    blockIndexPrev17.nTime = 1408728756;  // Block #123439
-    blockIndexPrev17.nBits = 0x1b118d9c;
-    blockIndexPrev16.pprev = &blockIndexPrev17;
-    CBlockIndex blockIndexPrev18 = CBlockIndex();
-    blockIndexPrev18.nTime = 1408728744;  // Block #123438
-    blockIndexPrev18.nBits = 0x1b11abac;
-    blockIndexPrev17.pprev = &blockIndexPrev18;
-    CBlockIndex blockIndexPrev19 = CBlockIndex();
-    blockIndexPrev19.nTime = 1408728608;  // Block #123437
-    blockIndexPrev19.nBits = 0x1b11951e;
-    blockIndexPrev18.pprev = &blockIndexPrev19;
-    CBlockIndex blockIndexPrev20 = CBlockIndex();
-    blockIndexPrev20.nTime = 1408728495;  // Block #123436
-    blockIndexPrev20.nBits = 0x1b121cf3;
-    blockIndexPrev19.pprev = &blockIndexPrev20;
-    CBlockIndex blockIndexPrev21 = CBlockIndex();
-    blockIndexPrev21.nTime = 1408728479;  // Block #123435
-    blockIndexPrev21.nBits = 0x1b11a33c;
-    blockIndexPrev20.pprev = &blockIndexPrev21;
-    CBlockIndex blockIndexPrev22 = CBlockIndex();
-    blockIndexPrev22.nTime = 1408728332;  // Block #123434
-    blockIndexPrev22.nBits = 0x1b10e09e;
-    blockIndexPrev21.pprev = &blockIndexPrev22;
-    CBlockIndex blockIndexPrev23 = CBlockIndex();
-    blockIndexPrev23.nTime = 1408728124;  // Block #123433
-    blockIndexPrev23.nBits = 0x1b104be1;
-    blockIndexPrev22.pprev = &blockIndexPrev23;
+    // Construct a chain of block index entries
+    std::list<CBlockIndex> blockidx;
```

Comment:
why `std::list`? let's rewrite it with `std::vector`

Reply:
I don't see any benefit to using `std::vector` and `std::list` is used by another test ([source](https://github.com/dashpay/dash/blob/d754799580d2914a0dd6388c90100d14feec0428/src/test/skiplist_tests.cpp#L147)), I think it's fine as-is.

Reply:
that's a single unit test, not a hot function,so, extra memory allocation is not important for performance. Agree, std::list is fine